asyncpg/types.py

```diff
@@ -85,6 +85,52 @@ def upper_inf(self):
     def isempty(self):
         return self._empty
 
+    def issubset(self, other):
+        if self._empty:
+            return True
+        if other._empty:
+            return False
+
+        if other._lower is None:
+            lowerOk = True
+        else:
+            if self._lower is None:
+                return False
+            if other._lower_inc or not self._lower_inc:
+                lowerOk = self._lower >= other._lower
+            else:
+                lowerOk = self._lower > other._lower
+
+        if not lowerOk:
+            return False
+
+        if other._upper is None:
+            upperOk = True
+        else:
+            if self._upper is None:
+                return False
+            if other._upper_inc or not self._upper_inc:
+                upperOk = self._upper <= other._upper
+            else:
+                upperOk = self._upper < other._upper
+
+        return lowerOk and upperOk
+
+    def issuperset(self, other):
+        return other.issubset(self)
+
+    def size(self, step=None):
+        if self._upper is None or self._lower is None:
+            return None
```

Comment:
It would be better to raise a `ValueError` here. `size()` on an open range is meaningless and akin to division by zero, returning `None` in this case risks masking bugs.

Reply:
As I understood it, a None in either lower/upper means infinity. Hence why I return the "infinity". 

For numbers this could be changed from None to `float("inf")`, and for datetimes to `datetime.datetime.min/max`.

Reply:
Since postgres only has numeric and date/time ranges adding type check for datetime and numeric and returning values for each one and removing None completely might be worthwhile refactor.